lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -4540,8 +4540,18 @@ instance IsServerError ErrBalanceTx where
                 , "makes no sense, and I'm confused."
                 ]
         ErrBalanceTxExistingCollateral ->
-            apiError err403 CreatedInvalidTransaction $ mconcat
-                [ "I cannot balance transactions with pre-defined collateral."
+            apiError err403 CreatedInvalidTransaction
+                "I cannot balance transactions with pre-defined collateral."
+
+        ErrBalanceTxExistingTotalCollateral ->
+            apiError err403 CreatedInvalidTransaction $ T.unwords
+                [ "I cannot balance transactions"
+                , "with pre-defined total collateral."
+                ]
+        ErrBalanceTxExistingReturnCollateral ->
+            apiError err403 CreatedInvalidTransaction $ T.unwords
```

Comment:
I think we have come to use `CreatedInvalidTransaction` everywhere in `IsServerError ErrBalanceTx` from the beginning just by accident. It might be better to break the pattern now and create new `ApiErrorCode` constructors.

Reply:
🆗